### ISSUE TITLE
Logout and token revocation

### ISSUE THREAD

**Issue by lamuertepeluda** [FusionAuth/fusionauth-issues#465]
### Description
Maybe I got it all wrong, but I am trying to understand if the Oauth2 logout would trigger a JWT refresh token revocation event (and call a webhook on my server).

From my attempts, it never gets called. Maybe this is just the correct behavior and my expectations are wrong.

I have the following scenario, that is 1 tenant with 1 application:
- A backend server (node, or .NET)
- A SPA Client which uses the Implicit Flow (HTML5/JS/React)

I would like to have a complete user logout with invalidation of both his/her refresh token and JWT (which, I know, may last a short while after a logout).

Therefore on the backend I have implemented the webhook following this blog post https://fusionauth.io/learn/expert-advice/tokens/revoking-jwts (JWT Manage and all). I have also connected the webhook for jwt revocation events and it works from when pushing the test button.

On the frontend I have a simple SPA wich uses https://www.npmjs.com/package/react-oauth2-hook for login.

The login is fine, but the logout with https://fusionauth.io/docs/v1/tech/oauth/endpoints#logout wouldn't work if I don't call the session delete from within the spa. The webhook is also never triggered (my server can never know if the SPA user ever did a logout).

In order to trigger the event and the webhook I have to call my backend API (which calls https://fusionauth.io/docs/v1/tech/apis/login#logout-a-user). But then what's the oauth logout endpoint for?

Maybe I just didn't got 100% how these things work/are related, or maybe this is just a missing feature?

What is the relationship between https://fusionauth.io/docs/v1/tech/apis/login#logout-a-user and 
https://fusionauth.io/docs/v1/tech/oauth/endpoints#logout?

*FusionAuth Version 1.14.0* (same with a few previous versions), from docker containers.

**Comment by robotdan** [comment]
We do not currently support OIDC back channel logout (which would revoke tokens), only front channel. 

https://openid.net/specs/openid-connect-backchannel-1_0.html
https://openid.net/specs/openid-connect-frontchannel-1_0.html

Outside of OIDC you can use the JWT APIs as part of your logout procedure. 
https://fusionauth.io/docs/v1/tech/apis/jwt#revoke-refresh-tokens

**Comment by lamuertepeluda** [comment]
@robotdan thank you very much for the references! 😃 

Is it possible that the OIDC backchannel will be ever supported by FusionAuth in the future?

**Comment by robotdan** [comment]
@lamuertepeluda Yes - if we get enough requests for it, this is definitely something we'd implement. 

In addition to the JWT API I referenced, you can also use our Logout API which can revoke a refresh token.  https://fusionauth.io/docs/v1/tech/apis/login#logout-a-user

I'll convert this issue to a feature and let the community vote.

**Comment by JuliusPC** [comment]
Using and _enforcing_ Back-Channel Logout improves protection from CSRF attacks (e.g. by tricking an user clicking on a link pointing to the Logout endpoint -- Yes, that is not dangerous per se).
